internal/pkg/cli/svc_deploy.go

```diff
@@ -395,6 +396,13 @@ func workloadManifest(in *workloadManifestInput) (manifest.DynamicWorkload, erro
 }
 
 func validateWorkloadManifestCompatibilityWithEnv(ws wsEnvironmentsLister, env versionCompatibilityChecker, mft manifest.DynamicWorkload, envName string) error {
+	currVersion, err := env.Version()
+	if err != nil {
+		return fmt.Errorf("get environment %q version: %w", envName, err)
+	}
```

Comment:
Do we have any super old environment template that doesn't have version?

Reply:
Good call. I think it is ok because we'll return `deploy.LegacyEnvTemplateVersion` which is `v0.0.0`